src/call/server.rs

```diff
@@ -367,23 +375,37 @@ macro_rules! impl_stream_sink {
                 };
 
                 $ft {
-                    call: self.call,
+                    call: self.call.take().unwrap(),
                     fail_f: fail_f,
                     err: err,
                 }
             }
         }
 
+        impl<T> Drop for $t<T> {
+            /// The corresponding RPC will be canceled if the sink did not call
+            /// [`close`] or [`fail`] before dropping.
+            ///
+            /// [`close`]: #method.close
+            /// [`fail`]: #method.fail
+            fn drop(&mut self) {
+                // We did not close it explicitly and it was not dropped in the `fail`.
+                if self.flush_f.is_none() && self.call.is_some() {
```

Comment:
why check flush here?

Reply:
Check if we have closed it.

Reply:
got it

Reply:
I think you can set a flag explicitly at L447.